src/Host.cpp

```
@@ -2598,3 +2628,956 @@ void Host::setCompactInputLine(const bool state)
+            dockwidget->setFloating(false);
+            mudlet::self()->addDockWidget(Qt::BottomDockWidgetArea, dockwidget);
+            return {true, QString()};
+        } else {
+            return {false, QStringLiteral(R"("docking option "%1" not available. available docking options are "t" top, "b" bottom, "r" right, "l" left and "f" floating")").arg(area)};
+        }
+    }
+}
+
+std::pair<bool, QString> Host::createMiniConsole(const QString& windowname, const QString& name, int x, int y, int width, int height)
+{
+    if (!mpConsole) {
+        return {false, QString()};
+    }
+
+    auto pC = mpConsole->mSubConsoleMap.value(name);
+    auto pW = mpConsole->mDockWidgetMap.value(name);
+    if (!pC) {
+        pC = mpConsole->createMiniConsole(windowname, name, x, y, width, height);
+        if (pC) {
+            pC->setFontSize(12);
+            return {true, QString()};
+        }
+    } else if (pC) {
+        // CHECK: The absence of an explict return statement in this block means that
+        // reusing an existing mini console causes the lua function to seem to
+        // fail - is this as per Wiki?
+        // This part was causing problems with UserWindows
+        if (!pW) {
+            pC->resize(width, height);
+            pC->move(x, y);
+            return {false, QStringLiteral("miniconsole \"%1\" exists already. moving/resizing \"%1\".").arg(name)};
+        }
+    }
+    return {false, QStringLiteral("miniconsole/userwindow \"%1\" exists already.").arg(name)};
+}
+
+std::pair<bool, QString> Host::createLabel(const QString& windowname, const QString& name, int x, int y, int width, int height, bool fillBg, bool clickthrough)
+{
+    if (!mpConsole) {
+        return {false, QString()};
+    }
+
+    auto pL = mpConsole->mLabelMap.value(name);
+    auto pC = mpConsole->mSubConsoleMap.value(name);
+    if (!pL && !pC) {
+        pL = mpConsole->createLabel(windowname, name, x, y, width, height, fillBg, clickthrough);
+        if (pL) {
```

Slightly erroneous logic here - if `TConsole::createLabel(...)` CAN fail in the prior line then we need to return an error message instead of falling through to the end of the function that returns a failure indication but no failure details. Then, if there is an unconditional return *with* an error message after THIS `if` then the `else if` for `pL` and `pC` become simpler `if`s and in fact the second (`if (pC)`) actually becomes an unconditional return "miniconsole/userwindow(**/buffer**) with this name exists" and the very last return is eliminated...

---

Actually currently `createLabel` cannot fail here because we already tested for its sole failure condition.

I'll rewrite that to clarify.